### DISCUSSION TITLE
How to collaborate?

### DISCUSSION THREAD

**Discussion by shoooe** [ShafSpecs/remix-pwa#158] [Q&A]
I have a couple of ideas for improvements on the library and I forked the project but I can't find a guide on how to setup the project and run the playground. Is there a guide somewhere that I'm missing? If not, what steps should I follow to setup my local environment?

Thanks.

**Comment by ShafSpecs** [comment]
The docs have a contributing guide: https://remix-pwa.run/docs/main/contributing

Regarding the sandbox, you would just run npm run dev to start it up. It isn't cleaned up, as I am the only one that tend to use it. Fork from dev if you also want to include Remix PWA v5 features. Main is currently v4

**Comment by shoooe** [comment]
@ShafSpecs  No yeah I read that. What I meant was a basic list of commands that I should run to get started. For example I tried to run `npm install` but I get:

```
npm error code ETARGET
npm error notarget No matching version found for @remix-pwa/manifest@3.0.0.
npm error notarget In most cases you or one of your dependencies are requesting
npm error notarget a package version that doesn't exist.
```

I'm assuming this is because we are using workspaces. I'm only familiar with workspaces in `yarn` and `bun` and there you could just `install` from the root so I'm a lil confused.

**Reply by ShafSpecs** [reply]
This should not happen since it is meant to install manifest from local, would take a look soon 👍

**Reply by ShafSpecs** [reply]
Fixed this. The script was missing a pre-run script

**Comment by shoooe** [comment]
I also tried to run `npm run postclone` as described here: https://github.com/remix-pwa/remix-pwa/blob/main/CONTRIBUTING.md
but I got the same error.

All of this ^ on `dev`.

On `main` instead if I run` npm run postclone` I get:

```
npm error code 1
npm error path /Users/shoe/dev/remix-pwa-monorepo/playground
npm error command failed
npm error command sh -c echo "Updating @remix-pwa/* packages" && node ../scripts/update-sandbox.js
npm error Updating @remix-pwa/* packages
npm error node:internal/errors:496
npm error     ErrorCaptureStackTrace(err);
npm error     ^
npm error
npm error TypeError [ERR_IMPORT_ASSERTION_TYPE_MISSING]: Module "file:///Users/shoe/dev/remix-pwa-monorepo/playground/package.json" needs an import assertion of type "json"
npm error     at new NodeError (node:internal/errors:405:5)
npm error     at validateAssertions (node:internal/modules/esm/assert:94:15)
npm error     at defaultLoad (node:internal/modules/esm/load:122:3)
npm error     at ModuleLoader.load (node:internal/modules/esm/loader:388:13)
npm error     at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:270:56)
npm error     at new ModuleJob (node:internal/modules/esm/module_job:65:26)
npm error     at #createModuleJob (node:internal/modules/esm/loader:282:17)
npm error     at ModuleLoader.getJobFromResolveResult (node:internal/modules/esm/loader:240:34)
npm error     at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:221:17)
npm error     at async ModuleLoader.import (node:internal/modules/esm/loader:307:23) {
npm error   code: 'ERR_IMPORT_ASSERTION_TYPE_MISSING'
npm error }
npm error
npm error Node.js v20.7.0
```

**Reply by ShafSpecs** [reply]
This error is due to Node 22 incompatibility. Importing json in node 21 and below happened with the assert keyword, in node 22, it is via the with keyword. And it's a breaking change. You would need nvm and update your node to 22 within the project

**Reply by ShafSpecs** [reply]
I've fixed this issue. Meaning postclone should now work 👍

**Comment by ShafSpecs** [comment]
You should now be able to fork the dev branch and make changes there, I am working on the sandbox itself, as there are two possible apps: SPA and SSR Remix applications
